### PR TITLE
install external packages for tests

### PR DESCRIPTION
**Pull request by orbeckst**
Proof of concept for #898

Changes made in this Pull Request:

Install additional packages for the **full** testing environment on Travis CI:
- install HOLE
- install clustalw2

This enables additional tests that are currently skipped because executables are not available.

PR Checklist
------------
 - n/a Tests?
 - n/a Docs?
 - [x] CHANGELOG updated? 
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
Works for `hole` in 74373ac (see e.g. Linux travis build [2104.2](https://travis-ci.org/MDAnalysis/mdanalysis/jobs/144029758)).

This works for Linux and OSX (see [2104.6](https://travis-ci.org/MDAnalysis/mdanalysis/jobs/144029764)).

The [HOLE](http://www.smartsci.uk/hole/) package is downloaded and unpacked into `${HOME}/opt/packages` and the bin directory added to `PATH`. Coverage of the `analysis.hole` module is now 59% (if we just *run* the plotting code it would be most of it...) and overall coverage +1.4% :-). 

Download and installation of `hole` is less than 2 s of the build time.

**Comment by orbeckst** [comment]
conda packages for [clustalw2](http://clustal.org) were built by the nice folks at https://biobuilds.org/ — they have both Linux and OSX conda packages:  https://anaconda.org/biobuilds/clustalw. It installs `clustalw2`, which is the name of the executable as present in the tests for `aanalysis.align`.

This was added in 642a1d9 and is currently running on Travis.

**Comment by orbeckst** [comment]
Needs opinions/reviews.

**Comment by orbeckst** [comment]
On OSX full `test_hole_module_fd_closure()` failed in [2106.6](https://travis-ci.org/MDAnalysis/mdanalysis/jobs/144037457) with 
```
ERROR: MDAnalysis.analysis.hole: Issue 129: ensure low level file descriptors to PDB files used by Hole program are properly closed
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/MDAnalysis/mdanalysis/miniconda/envs/pyenv/lib/python2.7/site-packages/numpy/testing/decorators.py", line 147, in skipper_func
  File "/Users/travis/build/MDAnalysis/mdanalysis/miniconda/envs/pyenv/lib/python2.7/site-packages/numpy/testing/decorators.py", line 147, in skipper_func
  File "/Users/travis/build/MDAnalysis/mdanalysis/miniconda/envs/pyenv/lib/python2.7/site-packages/MDAnalysisTests/analysis/test_hole.py", line 150, in test_hole_module_fd_closure
  File "/Users/travis/build/MDAnalysis/mdanalysis/miniconda/envs/pyenv/lib/python2.7/site-packages/MDAnalysisTests/tempdir.py", line 72, in __enter__

OSError: '[Errno 24] Too many open files\n-------------------- >> begin captured logging << --------------------\nMDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /Users/travis/build/MDAnalysis/mdanalysis/miniconda/envs/pyenv/lib/python2.7/site-packages/MDAnalysisTests/data/1grm_elNemo_mode7.pdb.bz2...\n--------------------- >> end captured logging << ---------------------'
...
```

The problems seems to be that there are already too many files open. Once we temporarily reduce the open file descriptors to 64 we cannot even create the tempdir anymore so the tests errors.

File: .travis.yml [hunk 1]
```diff
@@ -37,14 +38,28 @@ install:
...
   - chmod +x testsuite/MDAnalysisTests/mda_nosetests
+# additional external tools (Issue #898) -- HOLE
+  - |
+     if [[ $SETUP == 'full' ]]; then \
+         bash ./maintainer/install_hole.sh $TRAVIS_OS_NAME "${HOME}/${MDA_OPTPACKAGES}"; \
+         HOLE_BINDIR="${HOME}/${MDA_OPTPACKAGES}/hole2/exe"; \
+         export PATH=${PATH}:${HOLE_BINDIR}; \
+     fi
+  # we should still be here but make doubly sure:
+  - cd ${TRAVIS_BUILD_DIR}
```
Comment (jbarnoud) [inline .travis.yml]: Is there a reason to make doubly sure here specifically?

File: maintainer/install_hole.sh [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+esac
+
+PREFIX="$2"
+test -n "$PREFIX" || die "missing second argument PREFIX (installation directory)" 10
+
+#------------------------------------------------------------
+# start installation
+#------------------------------------------------------------
+
+cd ~
```
Comment (jbarnoud) [inline maintainer/install_hole.sh]: I am not sure I like a script that mess up with my home directory.

File: .travis.yml (hunk 1 above)
Comment (orbeckst) [inline .travis.yml]: I tried to make it a bit future-proof: You can put other installation scripts into this `if` statement. But if you are jumping directories then thr rest of the travis script will fail. The "making doubly sure" serves as a reminder that one of the assumptions of the travis script is that everything happens from within the checked-out git repo.

File: maintainer/install_hole.sh (hunk 2 above)
Comment (orbeckst) [inline maintainer/install_hole.sh]: I think that line is a left-over from an earlier version of the script when I hadn't switched over to using `PREFIX`. I think I can remove it.

In any case, it only has to work on travis...

Comment (orbeckst) [inline maintainer/install_hole.sh]: removed the `cd ~` line

**Comment by orbeckst** [comment]
@jbarnoud or @richardjgowers – do you want to review the PR?

**Comment by orbeckst** [comment]
I removed f02dc88 from the PR (and force-pushed). I thought I would switch coverage reporting to the second Travis job (`SETUP == full`) but that was not the case. (I only switched doc building to the `full` job but it's not clear that this is of any advantage.)

**Comment by orbeckst** [comment]
I do have a question, though: Which coverage badge is shown? When you look at https://coveralls.io/builds/7076831 you see 
- 79.77% 2118.1 (SETUP=minimal PYTHON_VERSION=2.7)
- 82.47% 2118.2 (SETUP=full PYTHON_VERSION=2.7)

I would like to show/see the one for job 2. How does one do that?

**Comment by jdetle** [comment]
I see the greater of the two. I have no idea why though.

**Comment by jdetle** [comment]
In my experience it shows the greater of the two. I remember this anecdotally because of the last PR I made increasing RMSD test coverage.

**Comment by orbeckst** [comment]
This passes (except the OS X full build, as discussed above). Note that the coverage for full increased by 2% to 83% :-).

**Comment by orbeckst** [comment]
I'll rebase and compact... and increase the number of file descriptors so that the OS X test passes... maybe checking for OS X.

**Comment by richardjgowers** [comment]
It's only failing on macos, I think the highest number of open files I saw was 250!

**Comment by orbeckst** [comment]
I added a hack to the test for OS X just to get it to pass. Currently the max number of open fd for Darwin is 128 but we can adjust it to say 256. It's a horrible fudge but we can revisit after the #363 merge. 

Or just skip the test on OS X?

--
Oliver Beckstein
email: orbeckst@gmail.com

**Comment by richardjgowers** [comment]
I'd rather fudge the fd limit for now.

**Comment by orbeckst** [comment]
Ok, will do that during the day (unless anyone else beats me to it… could do this with edit on GH…)

> On 20 Jul, 2016, at 09:48, Richard Gowers <notifications@github.com> wrote:
> 
> I'd rather fudge the fd limit for now.

--
Oliver Beckstein * orbeckst@gmx.net
skype: orbeckst  * orbeckst@gmail.com

**Comment by orbeckst** [comment]
Set the OS X fd limits to 256 and force-pushed.

**Comment by orbeckst** [comment]
Well, os x still fails. I'll try again with 512 open fd...

**Comment by orbeckst** [comment]
Tadaa! Take that, OS X!

@richardjgowers , this seems to be done. I would prefer if the commits were not squashed because they all do different little things and in the future it might come in handy to re-evaluate them individually.

**Comment by richardjgowers** [comment]
@orbeckst awesome, creeping back towards 90% coverage!

**Comment by orbeckst** [comment]
> On 20 Jul, 2016, at 14:57, Richard Gowers <notifications@github.com> wrote:
> 
> @orbeckst awesome, creeping back towards 90% coverage!

Partly by running Something.plot() and just testing that we can get a proper matplotlib axes instance… but yeah, I am pleased with +2% :-).

Once we throw stuff into legacy and remove it from coverage (how does one do that?) it will *look* even better ;-).

--
Oliver Beckstein * orbeckst@gmx.net
skype: orbeckst  * orbeckst@gmail.com

**Comment by jdetle** [comment]
We can get another big boost if we add coverage for PSA.

**Comment by orbeckst** [comment]
> On 20 Jul, 2016, at 15:00, John Detlefs <notifications@github.com> wrote:
> 
> We can get another big boost if we add coverage for PSA.

Ping @sseyler ;-) – maybe you can get @tcolburn to write some tests?

**Pull request merged**